lib/network/http2_network_handler.go

```diff
@@ -92,7 +90,9 @@ func BallotHandler(ctx context.Context, t *HTTP2Network) HandlerFunc {
 			http.Error(w, "Error reading request body", http.StatusInternalServerError)
 		}
 
-		t.ReceiveChannel() <- Message{Type: BallotMessage, Data: body}
+		re.ReceiveMessage(t, Message{Type: BallotMessage, Data: body})
```

Comment:
The `ReceiveChannel` method is intended not to call in test code so, re.ReceiveMessage function is called.